verification/curator-service/api/src/controllers/auth.ts

```diff
@@ -158,59 +208,148 @@ export class AuthController {
             },
         );
 
+        /**
+         * Generate reset password link
+         */
         this.router.post(
-            '/signup',
-            async (req: Request, res: Response): Promise<void> => {
-                let user = await User.findOne({
-                    email: req.body.email,
-                }).exec();
+            '/request-password-reset',
+            async (req: Request, res: Response): Promise<Response<any>> => {
+                const email = req.body.email as string;
 
-                // Update newsletter preferences
-                if (
-                    user &&
-                    user.newsletterAccepted === false &&
-                    req.body.newsletter
-                ) {
-                    user = await User.findOneAndUpdate(
-                        { email: req.body.email },
-                        { newsletterAccepted: req.body.newsletter },
+                try {
+                    // Check if user with this email address exists
+                    const user = await User.findOne({ email });
+                    if (!user) {
+                        return res.sendStatus(200);
+                    }
+
+                    // Check if user is a Gmail user and send appropriate email message in that case
+                    if (user.googleID !== '42') {
```

Comment:
The '42' is a hard-coded value from the test suite. I think the correct behaviour should be `if (user.googleID)` (it can't be `if (googleID in user)` because mongoose/mongo defines keys that are in the schema even if the document doesn't have a value).

Reply:
Fixed, thank you